tests/integration/test_agentd/test_agentd_state.py

```diff
@@ -0,0 +1,337 @@
+                return True
+    return False
+
+
+def check_last_keepalive(expected_value=None, get_state_callback=None):
+    """Check `field` status
+
+    Args:
+        expected_value (string, optional): value to check against.
+                                           Defaults to None.
+        get_state_callback (function, optional): callback to get state.
+                                                 Defaults to None.
+
+    Returns:
+        boolean: `True` if check was successfull. `False` otherwise
+    """
+    if get_state_callback:
+        current_value = get_state_callback()['last_keepalive']
+        if expected_value == '':
+            return expected_value == current_value
+
+    keep_alive_msg = 'Sending keep alive'
+    agent_notification_msg = 'Sending agent notification'
+
+    with open(LOG_FILE_PATH, 'r') as log:
+        for line in log:
+            if current_value.replace('-', '/') in line and (keep_alive_msg in line or agent_notification_msg in line):
+                return True
+    return False
+
+
+def check_msg_count(expected_value=None, get_state_callback=None):
+    """Check `field` status
+
+    Args:
+        expected_value (string, optional): value to check against.
+                                           Defaults to None.
+        get_state_callback (function, optional): callback to get state.
+                                                 Defaults to None.
+
+    Returns:
+        boolean: `True` if check was successfull. `False` otherwise
+    """
+    if get_state_callback:
+        current_value = get_state_callback()['msg_count']
+        if expected_value == '':
+            return expected_value == current_value
+
+    sent_messages = 0
+
+    with open(LOG_FILE_PATH, 'r') as log:
+        for line in log:
+            if 'Sending keep alive' in line:
+                sent_messages += 1
+
+    return sent_messages >= current_value
+
+
+def check_status(expected_value=None, get_state_callback=None):
+    """Check `field` status
+
+    Args:
+        expected_value (string, optional): value to check against.
+                                           Defaults to None.
+        get_state_callback (function, optional): callback to get state.
+                                                 Defaults to None.
+
+    Returns:
+        boolean: `True` if check was successfull. `False` otherwise
+    """
+    if expected_value != 'pending':
+        wait_keepalive(True)
+        if get_state_callback == parse_state_file:
+            wait_state_update(True)
+    current_value = get_state_callback()['status']
+    return expected_value == current_value
+
+
+def wait_connect(update_position=False):
+    """ Watch ossec.conf until `callback_connected_to_server` is triggered
+
+    Args:
+        update_position (bool, optional): update position after reading.
+                                          Defaults to False.
+    """
+    global wazuh_log_monitor
```

Comment:
Is this `global` really necessary?

Reply:
Unnecessary `global`s were removed, thanks :+1: . Current documentation format is Google Style Python Docstrings and is aligned with the new documentation style requirement.